RecoLocalTracker/SiPixelRecHits/src/PixelCPEFast.cc

```diff
@@ -185,21 +192,103 @@ void PixelCPEFast::fillParamsForGpu() {
     LogDebug("PixelCPEFast") << i << ' ' << m * std::sqrt(lape.xx()) << ' ' << m * std::sqrt(lape.yy());
 #endif  // EDM_ML_DEBUG
 
-    errorFromTemplates(p, cp, 20000.f);
     g.pixmx = std::max(0, cp.pixmx);
-    g.sx[0] = cp.sigmax;
-    g.sx[1] = cp.sx1;
-    g.sx[2] = cp.sx2;
+    g.sx2 = toMicron(cp.sx2);
+    g.sy1 = std::max(21, toMicron(cp.sy1));  // for some angles sy1 is very small
+    g.sy2 = std::max(55, toMicron(cp.sy2));  // sometimes sy2 is smaller than others (due to angle?)
+
+    // sample xerr as function of position
+    auto xoff = -81.f * commonParamsGPU_.thePitchX;
+
+    for (int ix = 0; ix < 16; ++ix) {
+      auto x = xoff + (0.5f + float(ix)) * 162.f * commonParamsGPU_.thePitchX / 16.f;
+      auto gvx = p.theOrigin.x() - x;
+      auto gvy = p.theOrigin.y();
+      auto gvz = 1.f / p.theOrigin.z();
+      cp.cotbeta = gvy * gvz;
+      cp.cotalpha = gvx * gvz;
+      errorFromTemplates(p, cp, 20000.f);
+      g.sigmax[ix] = toMicron(cp.sigmax);
+      g.sigmax1[ix] = toMicron(cp.sx1);
+#ifdef EDM_ML_DEBUG
+      LogDebug("PixelCPEFast") << "sigmax vs x " << i << ' ' << x << ' ' << cp.cotalpha << ' ' << int(g.sigmax[ix])
+                               << ' ' << int(g.sigmax1[ix]) << ' ' << 10000.f * cp.sigmay << std::endl;
+#endif  // EDM_ML_DEBUG
```

Comment:
```suggestion
      LogDebug("PixelCPEFast") << "sigmax vs x " << i << ' ' << x << ' ' << cp.cotalpha << ' ' << int(g.sigmax[ix])
                               << ' ' << int(g.sigmax1[ix]) << ' ' << 10000.f * cp.sigmay << std::endl;
```
ifdef seems redundant here